traffic_portal/app/src/common/api/CDNService.js

```diff
@@ -17,252 +17,199 @@
-                },
-                function(fault) {
-                    request.reject();
-                }
-            );
-
-        return request.promise;
+       return $http.get(ENV.api['root'] + 'cdns/' + cdnName + '/snapshot').then(
+            function(result) {
+                return result.data.response;
+            },
+            function(err) {
+                console.error(err);
+            }
+        );
     };
 
     this.getNewSnapshot = function(cdnName) {
-        var request = $q.defer();
-
-        $http.get(ENV.api['root'] + "cdns/" + cdnName + "/snapshot/new")
-            .then(
-                function(result) {
-                    request.resolve(result.data.response);
-                },
-                function(fault) {
-                    request.reject();
-                }
-            );
-
-        return request.promise;
+        return $http.get(ENV.api['root'] + 'cdns/' + cdnName + '/snapshot/new').then(
+            function(result) {
+                return result.data.response;
+            },
+            function(err) {
+                console.error(err);
+            }
+        );
     };
 
     this.snapshot = function(cdn) {
-        var request = $q.defer();
-
-        $http.put(ENV.api['root'] + "cdns/" + cdn.id + "/snapshot")
-            .then(
-                function() {
-                    messageModel.setMessages([ { level: 'success', text: 'Snapshot performed' } ], true);
-                    locationUtils.navigateToPath('/cdns/' + cdn.id);
-
-                },
-                function(fault) {
-                    messageModel.setMessages(fault.data.alerts, false);
-                }
-            );
-
-        return request.promise;
+        return $http.put(ENV.api['root'] + 'cdns/' + cdn.id + '/snapshot').then(
+            function() {
+                messageModel.setMessages([{level: 'success', text: 'Snapshot performed'}], true);
+                locationUtils.navigateToPath('/cdns/' + cdn.id);
+            },
+            function(err) {
+                messageModel.setMessages(err.data.alerts, false);
+            }
+        );
     };
 
     this.getDNSSECKeys = function(cdnName) {
-        var request = $q.defer();
-
-        $http.get(ENV.api['root'] + "cdns/name/" + cdnName + "/dnsseckeys")
-            .then(
-                function(result) {
-                    request.resolve(result.data.response);
-                },
-                function(fault) {
-                    request.reject();
-                }
-            );
-
-        return request.promise;
+        return $http.get(ENV.api['root'] + 'cdns/name/' + cdnName + '/dnsseckeys').then(
+            function(result) {
+                return result.data.response;
+            },
+            function(err) {
+                console.error(err);
+            }
+        );
     };
 
     this.generateDNSSECKeys = function(dnssecKeysRequest) {
```

Comment:
this doesn't seem to work

https://localhost:8442/#!/cdns/2/dnssec-keys/generate and hit regenerate:

```
shared-libs.js?built=1558556756347:66530 TypeError: Cannot read property 'then' of undefined
    at generate (FormGenerateCdnDnssecKeysController.js:25)
```